modules/core/src/main/java/org/locationtech/jts/io/WKTReader.java

```diff
@@ -587,51 +889,12 @@ private MultiPoint OLDreadMultiPointText() throws IOException, ParseException {
    *@throws  IOException     if an I/O error occurs
    *@throws  ParseException  if an unexpected token was encountered
    */
-  private MultiPoint readMultiPointText() throws IOException, ParseException 
+  private MultiPoint readMultiPointText(StreamTokenizer tokenizer, EnumSet<Ordinate> ordinateFlags) throws IOException, ParseException
   {
-    String nextToken = getNextEmptyOrOpener();
-    if (nextToken.equals(EMPTY)) {
-      return geometryFactory.createMultiPoint(new Point[0]);
-    }
-    
-    // check for old-style JTS syntax and parse it if present
-  	// MD 2009-02-21 - this is only provided for backwards compatibility for a few versions
-  	if (ALLOW_OLD_JTS_MULTIPOINT_SYNTAX) {
-  		String nextWord = lookaheadWord();
-  		if (nextWord != L_PAREN) {
-  			return geometryFactory.createMultiPoint(toPoints(getCoordinatesNoLeftParen()));
-  		}
-  	}
-  	
-    ArrayList points = new ArrayList();
-    Point point = readPointText();
-    points.add(point);
-    nextToken = getNextCloserOrComma();
-    while (nextToken.equals(COMMA)) {
-    	point = readPointText();
-      points.add(point);
-      nextToken = getNextCloserOrComma();
-    }
-    Point[] array = new Point[points.size()];
-    return geometryFactory.createMultiPoint((Point[]) points.toArray(array));
+    return geometryFactory.createMultiPoint(
```

Comment:
@FObermaier This change causes the issue reported in #563.  Is there any reason reading EMPTY Points in a MULTIPOINT was removed?

Reply:
It was not removed on purpose. It was changed to streamline reading Coordinate[Sequence]s and use the factory methods that take CoordinateSequences as arguments.
The unit test present did not sugguest that a `MULTIPOINT` containing EMPTY points is a case to be considered.

Reply:
Fair enough.  The WKT handling of EMPTY is complex.  I will fix and add a unit test.